cmd/krel/cmd/release_notes.go

```diff
@@ -787,11 +899,393 @@ func (o *releaseNotesOptions) Validate() error {
 	}
 
 	// Options for PR creation
-	if releaseNotesOpts.createDraftPR || releaseNotesOpts.createWebsitePR {
-		if releaseNotesOpts.githubOrg == "" {
+	if o.createDraftPR || o.createWebsitePR {
+		if o.githubOrg == "" {
 			return errors.New("cannot generate the Release Notes PR without --org")
 		}
 	}
 
 	return nil
 }
+
+// Save the session to a file
+func (sd *sessionData) Save() error {
+	if sd.Date == 0 {
+		return errors.New("unable to save session, date is note defined")
+	}
+	if sd.Path == "" {
+		return errors.New("unable to save session, path is not defined")
+	}
+
+	jsonData, err := json.Marshal(sd)
+	if err != nil {
+		return errors.Wrap(err, "marshaling session data")
+	}
+
+	if err := ioutil.WriteFile(
+		fmt.Sprintf("%s%smaps-%d.json", sd.Path, string(os.PathSeparator), sd.Date),
+		jsonData, os.FileMode(0o644)); err != nil {
+		return errors.Wrap(err, "writing session data to disk")
+	}
+	return nil
+}
+
+// readFixSessions reads all the previous fixing data
+func readFixSessions(sessionPath string) (pullRequestChecklist map[int]bool, err error) {
+	files, err := ioutil.ReadDir(sessionPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading working directory")
+	}
+	pullRequestList := make([]int, 0)
+
+	// Look in the work dir for all json files
+	for _, fileData := range files {
+		currentSession := &sessionData{}
+		if strings.HasSuffix(fileData.Name(), ".json") {
```

Comment:
We assume that any `.json` file in the working directory is a session file?

Reply:
Yes all .json files in that directory are considered session files. It should not have other yaml or json files and they should not be allowed during the PR review.

Reply:
Okay. Maybe a debug level log line to just print out the name of loaded files?

Reply:
Added :+1: